src/artm/regularizer/decorrelator_phi.cc

```diff
@@ -19,12 +20,23 @@ bool DecorrelatorPhi::RegularizePhi(const ::artm::core::PhiMatrix& p_wt,
   // read the parameters from config and control their correctness
   const int topic_size = p_wt.topic_size();
   const int token_size = p_wt.token_size();
+  const bool use_topic_pairs = (topic_pairs_.size() > 0);
 
-  std::vector<bool> topics_to_regularize;
-  if (config_.topic_name().size() == 0)
-    topics_to_regularize.assign(topic_size, true);
-  else
-    topics_to_regularize = core::is_member(p_wt.topic_name(), config_.topic_name());
+  std::unordered_map<std::string, int> topics_to_regularize;
+  if (!use_topic_pairs) {
+    for (auto& s : (config_.topic_name().size() ? config_.topic_name() : p_wt.topic_name())) {
+      for (int i = 0; i < p_wt.topic_name().size(); ++i) {
+        if (p_wt.topic_name(i) == s)
+          topics_to_regularize.insert(std::make_pair(s, i));
+        else
+          LOG(WARNING) << "Topic name " << s << " is not presented into model and will be ignored";
```

Comment:
Is it ok that this warning appears for all indices `i` before break for each topic_name `s`?